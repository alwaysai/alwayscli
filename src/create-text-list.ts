import { regularizeText } from './util';

type Item = {
  name: string;
  text?: string;
};

function createTextListParagraph(item: Item, targetNameLength: number = 0) {
  const { name, text } = item;
  const lines = regularizeText(text).split('\n');
  const paddedName = name.padEnd(targetNameLength);
  let firstLine = paddedName;
  if (lines[0]) {
    firstLine += ` : ${lines[0]}`;
  }
  const padding = ''.padEnd(targetNameLength + 3);
  const paddedLines = lines.slice(1).map(line => `${padding}${line}`);
  return [firstLine, ...paddedLines].join('\n');
}

export function createTextList(...items: Item[]) {
  const targetNameLength = Math.max(...items.map(({ name }) => name.length));
  const paragraphs = items.map(item => createTextListParagraph(item, targetNameLength));
  return paragraphs.join('\n');
}
