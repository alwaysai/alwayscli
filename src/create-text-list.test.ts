import { createTextList } from './create-text-list';

describe(createTextList.name, () => {
  it('Creates a usage string for a branch', () => {
    const usageString = createTextList(
      { name: 'foo', text: 'hi\n   bye' },
      { name: 'barrio', text: 'bye' },
    );
    expect(usageString).toMatchSnapshot();
  });
});
