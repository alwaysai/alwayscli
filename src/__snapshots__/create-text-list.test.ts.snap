// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`createTextList Creates a usage string for a branch 1`] = `
"foo    : hi
            bye
barrio : bye"
`;
